java-checks/src/main/java/org/sonar/java/checks/ImmediatelyReturnedVariableCheck.java

```diff
@@ -79,12 +84,23 @@ public void visitBlock(BlockTree tree) {
             // Can only happen for non-compilable code, still, we should not report anything.
             return;
           }
-          context.reportIssue(
-            this, initializer, "Immediately " + lastTypeForMessage + " this expression instead of assigning it to the temporary variable \"" + identifier + "\".");
+          ((InternalJavaIssueBuilder) ((DefaultJavaFileScannerContext) context).newIssue())
+            .forRule(this)
+            .onTree(initializer)
+            .withMessage(String.format("Immediately %s this expression instead of assigning it to the temporary variable \"%s\".", lastTypeForMessage, identifier))
+            .withQuickFix(() -> quickFix(butLastStatement, lastStatement, variableTree, lastTypeForMessage))
+            .report();
         }
       }
     }
+  }
 
+  private static JavaQuickFix quickFix(StatementTree butLastStatement, StatementTree lastStatement, VariableTree variableTree, String lastTypeForMessage) {
+    // Equal token can not be null at this point, we checked before the presence of the initializer
+    return JavaQuickFix.newQuickFix("Inline expression")
+      .addTextEdit(JavaTextEdit.replaceBetweenTree(variableTree.modifiers(), variableTree.equalToken(), lastTypeForMessage),
```

Comment:
modifiers has been always emtpy in our examples. I would like to see one with `final`, such as:
```java
final int i = 42;
return i;
```
Also, I wonder about the end tree. What if the initializer is on a different line? or after many spaces?
Maybe it would be simpler to just replace from the beginning of the modifiers (if there is any) included until the beginning of the initializer, not included.

Reply:
Indeed, sounds cleaner to do it this way.